packages/garbo/src/resources/mayam.ts

```diff
@@ -0,0 +1,262 @@
+/* eslint-disable libram/verify-constants */
+import { cliExecute, Item, myLevel, useFamiliar } from "kolmafia";
+import { $item, clamp, get, maxBy, MayamCalendar } from "libram";
+import { garboValue } from "../garboValue";
+import { GarboTask } from "../tasks/engine";
+import getExperienceFamiliars from "../familiar/experienceFamiliars";
+import { globalOptions } from "../config";
+
+type MayamRingSummon = {
+  choice: string;
+  value: () => number;
+  ring: number;
+};
+
+type Resonance = {
+  rings: string;
+  item: Item;
+};
+
+const RESONANCE: Resonance[] = [
+  { rings: "eye yam2 eyepatch yam4", item: $item`Mayam spinach` },
+  { rings: "vessel yam2 cheese explosion", item: $item`stuffed yam stinkbomb` },
+  { rings: "yam1 meat cheese yam4", item: $item`yam and swiss` },
+  { rings: "yam1 lightning yam3 clock", item: $item`yam battery` },
+  { rings: "yam1 yam2 yam3 explosion", item: $item`thanksgiving bomb` },
+  // {rings: "yam1 meat eyepatch yam4", item: $item`yamtility belt`}
+];
```

Comment:
Several of these functions are made redundant by libram support. I would nix the redundant functions and update the type of `MayamRingSummon` to use a more precise typing rather than just "string"

Reply:
rather than doing `as [MayamSymbol, MayamSymbol, MayamSymbol, MayamSymbol]`, we need `as MayamCalendar.Combination`. I'll add some type-safe helper functions to `MayamCalendar` at some point, to make it so you don't have to cast the types